src/commands/CreateEnvironmentVariableCommand.ts

```diff
@@ -104,24 +113,50 @@ export function handleEnvironmentVariableCreation(dvConfig: DataVirtConfig, envi
 				resolve(false);
 			}
 		} else {
-			extension.log(`handleEnvironmentVariableCreation: Unable to create the environment variable ${entryName ? entryName : '<Unknown>'}...`);
+			extension.log(`handleEnvironmentVariableCreation: Unable to create the environment variable ${variableName ? variableName : '<Unknown>'}...`);
 			resolve(false);
 		}
 	});
 }
 
-function setEnvironmentVariableValue(entries: Property[], entryType: string, entryName: string, entryValue: string, refName?: string, refKey?: string): void {
-	let entry: Property;
-	if (entryType === constants.REFERENCE_TYPE_SECRET) {
-		const secretRef = new SecretRef(new KeyRef(refName, refKey));
-		entry = new Property(entryName, undefined, secretRef);
-	} else if (entryType === constants.REFERENCE_TYPE_CONFIGMAP) {
-		const configMapRef = new ConfigMapRef(new KeyRef(refName, refKey));
-		entry = new Property(entryName, undefined, configMapRef);
+function setEnvironmentVariableValue(entries: Property[], variableType: string, variableName: string, variableValue: string, refName?: string): void {
+	let variable: Property;
+	if (variableType === constants.REFERENCE_TYPE_SECRET) {
+		const secretRef = new SecretRef(new KeyRef(refName, variableName));
+		variable = new Property(variableName, undefined, secretRef);
+	} else if (variableType === constants.REFERENCE_TYPE_CONFIGMAP) {
+		const configMapRef = new ConfigMapRef(new KeyRef(refName, variableName));
+		variable = new Property(variableName, undefined, configMapRef);
+	} else {
+		variable = new Property(variableName, variableValue, undefined);
+	}
+	entries.push(variable);
+}
+
+async function queryVariable(envNode: EnvironmentTreeNode, predefinedVariables: Array<Property>): Promise<Property | undefined> {
+	const names: string[] = new Array<string>();
+	names.push(CREATE_NEW_ENTRY);
+
+	predefinedVariables.forEach( (variable: Property) => {
+		names.push(variable.name);
+	});
+
+	const selectedVariableName: string = await vscode.window.showQuickPick(names, { canPickMany: false, placeHolder: `Select a variable from the list or enter a new one` });
```

Comment:
shouldn't we have a validation on the name that can be provided?

Reply:
It is a quick pick. If those values are part of an existing ConfigMap or Secret I would consider them as valid. 
@rareddy Any opinion on that matter? Do we need an additional check for invalid Secret / ConfigMap contents not created by our DV extension?

Reply:
I am taking this discussion to the next dv tooling call. If required this will be addressed in a separate jira.